tests/test_state.py

```diff
@@ -0,0 +1,132 @@
+import json
+import os
+from unittest.mock import MagicMock
+
+import pytest
+
+import gepa.core.state as state_mod
+import gepa
+
+
+@pytest.fixture
+def run_dir(tmp_path):
+    os.makedirs(tmp_path / "run")
+    return tmp_path / "run"
+
+
+def test_initialize_gepa_state_fresh_init_writes_and_counts(run_dir):
+    """With a run dir but no state, the state is initialized from scratch and the eval output is written to the run dir."""
+    seed = {"model": "m"}
+    valset_out = (["out0", {"k": "out1"}], [0.1, 0.2])
+
+    fake_logger = MagicMock()
+    valset_evaluator = MagicMock(return_value=valset_out)
+
+    result = state_mod.initialize_gepa_state(
+        run_dir=str(run_dir),
+        logger=fake_logger,
+        seed_candidate=seed,
+        valset_evaluator=valset_evaluator,
+        track_best_outputs=False,
+    )
+
+    assert isinstance(result, state_mod.GEPAState)
+    assert result.num_full_ds_evals == 1
+    assert result.total_num_evals == len(valset_out[1])
+    fake_logger.log.assert_not_called()
+    valset_evaluator.assert_called_once_with(seed)
+
+    # Files written for each task with outputs (not scores)
+    base = run_dir / "generated_best_outputs_valset"
+    p0 = base / "task_0" / "iter_0_prog_0.json"
+    p1 = base / "task_1" / "iter_0_prog_0.json"
+    assert p0.exists() and p1.exists()
```

Comment:
The test should clean up the run_dir. Is that happening somewhere?

Reply:
no, because `run_dir` is created using the [`tmp_path` fixture](https://docs.pytest.org/en/stable/how-to/tmp_path.html#the-tmp-path-fixture) which is temporary to each test function and not retained